layout_riscv32.ld

```diff
@@ -1,9 +1,13 @@
 /* Layout for the RISC-V 32 boards, used by the examples in this repository. */
 
 MEMORY {
-  /* The TBF header region is 32 bytes (0x20) */
-  FLASH (rx) : ORIGIN = 0x20430020, LENGTH = 32M
-  SRAM (rwx) : ORIGIN = 0x80000000, LENGTH = 512K
+  /*
+   * The TBF header can change in size so use 0x40 combined with
+   * --protected-region-size with elf2tab to cover a header upto that
+   * size.
+   */
+  FLASH (rx) : ORIGIN = 0x20430040, LENGTH = 32M
+  SRAM (rwx) : ORIGIN = 0x80002400, LENGTH = 512K
```

Comment:
You don't mention why you are changing this addresses in the PR, can you explain why this is being changed.

Reply:
Ah that got dropped from the commit message. Sorry about that.

The short answer is both of these addresses are wrong. The header size is not actually fixed at 0x2c. The comment above explains why we want to use the flash offset that we do. The SRAM address is incorrect because we don't actually load at the first address of SRAM but at the address of APP_MEMORY. The address of SRAM may change depending on how you are building which is something I'd like to fix but for now we can at least add a better guess. I'll extend the comment to indicate that as well.